### PR TITLE
added usb controller

### PR DESCRIPTION
**Pull request by benadida**

### CODE REVIEW

File: src/lib/usbstick.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+export const isAvailable = () => {
+  return !!window.kiosk
+}
+
+export const isPresent = async () => {
+  return isAvailable() && (await window.kiosk!.getUsbDrives()).length > 0
+}
+
+const getDevice = async () => {
```
Comment (eventualbuddha) [inline src/lib/usbstick.ts:9]: This function is not safe to call in non-kiosk browser environments, but the others are.

File: src/lib/usbstick.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  return (await window.kiosk!.getUsbDrives())[0]
+}
+
+const isDeviceMounted = (device: KioskBrowser.UsbDrive) => {
```
Comment (eventualbuddha) [inline src/lib/usbstick.ts:13]: If `device` might not be set, do `device?: …`. You can also do `Boolean(device?.mountPoint)`.

File: src/lib/usbstick.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  if (!isAvailable()) {
+    return
+  }
+
+  const device = await getDevice()
+  if (isDeviceMounted(device)) {
+    return
+  }
+
+  window.kiosk!.mountUsbDrive(device.deviceName)
```
Comment (eventualbuddha) [inline src/lib/usbstick.ts:36]: `await` this?

File: src/lib/usbstick.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  if (!isAvailable()) {
+    return
+  }
+
+  const device = await getDevice()
+  if (!isDeviceMounted(device)) {
+    return
+  }
+
+  window.kiosk!.unmountUsbDrive(device.deviceName)
```
Comment (eventualbuddha) [inline src/lib/usbstick.ts:49]: `await` this?

**Comment by benadida** [comment]
I somehow reimplemented this in a separate PR #83 – which passes tests so I'm going to focus on that one, but take @eventualbuddha's comments into account.

**Comment by benadida** [comment]
addressed all issues in PR #83 , closing here.

**Pull request closed without merging**